### PR TITLE
feat(opentelemetry-instrumentation-aiokafka): wrap getone instead of anext, add tests

### PR DESCRIPTION
**Pull request by dimastbk**
# Description

Wrap getone instead of anext
Add tests for opentelemetry-instrumentation-aiokafka (#2082)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Does This PR Require a Core Repo Change?

- [ ] Yes.
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-aiokafka/tests/test_instrumentation.py [hunk 1]
```diff
@@ -11,15 +11,74 @@
...
+        producer._message_accumulator.add_message = mock.AsyncMock()
+        producer._sender.start = mock.AsyncMock()
+        producer._partition = mock.Mock(return_value=1)
+
+        await producer.start()
+
+        return producer
 
-class TestAIOKafka(TestCase):
     def test_instrument_api(self) -> None:
```
Comment (xrmx) [inline instrumentation/opentelemetry-instrumentation-aiokafka/tests/test_instrumentation.py:82]: I'm not sure you can keep async and not async tests on the same class.

File: instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/__init__.py [hunk 2]
```diff
@@ -126,10 +126,10 @@ def _instrument(self, **kwargs):
         )
         wrap_function_wrapper(
             aiokafka.AIOKafkaConsumer,
-            "__anext__",
-            _wrap_anext(tracer, async_consume_hook),
+            "getone",
```
Comment (xrmx) [inline instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/__init__.py:129]: Why are we changing the wrapped function?

File: instrumentation/opentelemetry-instrumentation-aiokafka/tests/test_instrumentation.py (hunk 1 above)
Comment (dimastbk) [reply instrumentation/opentelemetry-instrumentation-aiokafka/tests/test_instrumentation.py]: Hm. I've never encountered this, but I divided it into two test classes.

File: instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/__init__.py (hunk 2 above)
Comment (dimastbk) [reply instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/__init__.py:129]: __anext__ evaluate `getone` under the hood, and `getone` can be used directly.

**Comment by lzchen** [comment]
@dimastbk 

Would you mind adding yourself to [component_owners.yml](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/.github/component_owners.yml) according to [CONTRIBUTING.MD](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md#expectations-from-contributors)

**Comment by dimastbk** [comment]
@lzchen done

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `opentelemetry-instrumentation-fastapi` Add autoinstrumentation mechanism tests.
   ([#2860](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2860))
 - `opentelemetry-instrumentation-aiokafka` Add instrumentor and auto instrumentation support for aiokafka
-  ([#2082](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2082))
+  ([#2082](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2082), [#2874](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2874))
```
Comment (emdneto) [inline CHANGELOG.md:15]: I think you can add a new entry here

Comment (dimastbk) [reply CHANGELOG.md]: done

**Review by xrmx — APPROVED** [review]
It adds more tests than code so it can't be wrong

**Review by emdneto — APPROVED** [review]

**Pull request merged**